### ISSUE TITLE
BUG: Series.take needs to validate axis

### ISSUE THREAD

**Issue by jbrockmendel** [pandas-dev/pandas#51022]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
pd.Series([1, 2, 3]).take([1], axis="foo")
```

### Issue Description

Nonsense "axis" fails to raise

### Expected Behavior

ValueError

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

**Comment by aneesh98** [comment]
take

**Comment by aneesh98** [comment]
@jbrockmendel @lithomas1 Can you please review the linked pull request? [#51043 ]

**Comment by nikhilsharma26500** [comment]
Hi, is this issue still open?

**Comment by lithomas1** [comment]
Yes, but there is already an open pull request(there is a linter failure there which needs to be cleared).

I'd recommend looking at other good first issues first.

**Comment by nikhilsharma26500** [comment]
Alright, thanks for letting me know!

**Comment by Mall0-w** [comment]
take

**Comment by zgdjcls** [comment]
Hi, this is my first time using Pandas, may I ask if I added a new test case in [pandas/tests/series/indexing/test_take.py], how should I check if this error is fixed? Can I run test_take.py directly before and after the fix to verify this error has been solved?

**Comment by jbrockmendel** [comment]
run `pytest pandas/tests/series/indexing/test_take.py`

**Comment by zgdjcls** [comment]
Thank you, I will try it later

**Comment by abadrinath15** [comment]
@jbrockmendel I don't what to step on any toes but since the PR's gone stale quite a bit ago, would it be appropriate to tackle this? Happy to ensure your comments from prior contributor's pr are incorporated etc?

**Comment by abadrinath15** [comment]
take

**Comment by abadrinath15** [comment]
@jbrockmendel perhaps I'm missing something but this looks to be resolved? 

`
In [6]: pd.Series([1, 2, 3]).take([1], axis="foo")
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
File ~/Projects/pandas/pandas/core/generic.py:517, in NDFrame._get_axis_number(cls, axis)
    516 try:
--> 517     return cls._AXIS_TO_AXIS_NUMBER[axis]
    518 except KeyError:

KeyError: 'foo'

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
Cell In[6], line 1
----> 1 pd.Series([1, 2, 3]).take([1], axis="foo")

File ~/Projects/pandas/pandas/core/generic.py:3948, in NDFrame.take(self, indices, axis, **kwargs)
   3941     # We can get here with a slice via DataFrame.__getitem__
   3942     indices = np.arange(
   3943         indices.start, indices.stop, indices.step, dtype=np.intp
   3944     )
   3946 new_data = self._mgr.take(
   3947     indices,
-> 3948     axis=self._get_block_manager_axis(axis),
   3949     verify=True,
   3950 )
   3951 return self._constructor(new_data).__finalize__(self, method="take")

File ~/Projects/pandas/pandas/core/generic.py:537, in NDFrame._get_block_manager_axis(cls, axis)
    533 @final
    534 @classmethod
    535 def _get_block_manager_axis(cls, axis: Axis) -> AxisInt:
    536     """Map the axis to the block_manager axis."""
--> 537     axis = cls._get_axis_number(axis)
    538     ndim = cls._AXIS_LEN
    539     if ndim == 2:
    540         # i.e. DataFrame

File ~/Projects/pandas/pandas/core/generic.py:519, in NDFrame._get_axis_number(cls, axis)
    517     return cls._AXIS_TO_AXIS_NUMBER[axis]
    518 except KeyError:
--> 519     raise ValueError(f"No axis named {axis} for object type {cls.__name__}")

ValueError: No axis named foo for object type Series
`

**Comment by jbrockmendel** [comment]
Yep, this correctly raises on main.  Closing.
